File: apps/desktop/scripts/create-macos-icon.js
import sharp from 'sharp';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';
import { existsSync } from 'fs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const size = 1024;
const cornerRadius = size * 0.22; // ~22% corner radius (standard macOS)
const shadowOffset = 10;
const shadowBlur = 30;
const shadowOpacity = 0.3;

// Paths relative to script location
const inputPath = join(__dirname, '../../../images/radium-logo-square.png');
const outputPath = join(__dirname, '../app-icon.png');

console.log('Creating macOS-style icon...');
console.log(`Input: ${inputPath}`);
console.log(`Output: ${outputPath}`);

if (!existsSync(inputPath)) {
  console.error(`Error: Input file not found at ${inputPath}`);
  process.exit(1);
}

try {
  // Load and resize the source image
  const sourceImage = sharp(inputPath);
  const metadata = await sourceImage.metadata();
  
  console.log(`Source image: ${metadata.width}x${metadata.height}`);

  // Create rounded rectangle SVG mask
  const roundedRectSvg = Buffer.from(
    `<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
      <rect x="0" y="0" width="${size}" height="${size}" 
            rx="${cornerRadius}" ry="${cornerRadius}" 
            fill="white"/>
    </svg>`
  );

  // Create shadow layer
  const shadowSize = size + shadowOffset * 2;
  const shadowSvg = Buffer.from(
    `<svg width="${shadowSize}" height="${shadowSize}" xmlns="http://www.w3.org/2000/svg">
      <defs>
        <filter id="shadow">
          <feGaussianBlur in="SourceAlpha" stdDeviation="${shadowBlur / 2}"/>
        </filter>
      </defs>
      <rect x="${shadowOffset}" y="${shadowOffset}" 
            width="${size}" height="${size}" 
            rx="${cornerRadius}" ry="${cornerRadius}" 
            fill="black" 
            opacity="${shadowOpacity}"
            filter="url(#shadow)"/>
    </svg>`
  );

  // Process the image
  const resizedImage = await sourceImage
    .resize(size, size, { 
      fit: 'contain', 
      background: { r: 0, g: 0, b: 0, alpha: 0 } 
    })
    .toBuffer();

  // Create shadow
  const shadowBuffer = await sharp(shadowSvg)
    .resize(shadowSize, shadowSize)
    .png()
    .toBuffer();

  // Composite: shadow + image with rounded corners
  await sharp({
    create: {
      width: shadowSize,
      height: shadowSize,
      channels: 4,
      background: { r: 0, g: 0, b: 0, alpha: 0 }
    }
  })
    .composite([
      {
        input: shadowBuffer,
        blend: 'over'
      },
      {
        input: resizedImage,
        left: shadowOffset,
        top: shadowOffset,
        blend: 'over'
      },
      {
        input: roundedRectSvg,
        left: shadowOffset,
        top: shadowOffset,
        blend: 'dest-in'
      }
    ])
    .png()
    .toFile(outputPath);

  console.log(`✅ macOS icon created successfully at ${outputPath}`);
  console.log(`   Size: ${shadowSize}x${shadowSize}px`);
  console.log(`   Corner radius: ${cornerRadius}px`);
  console.log(`   Shadow: ${shadowOffset}px offset, ${shadowBlur}px blur`);
} catch (error) {
  console.error('Error creating icon:', error);
  process.exit(1);
}

File: apps/desktop/scripts/create-macos-icon.py
#!/usr/bin/env python3
"""
Create a macOS-style app icon with rounded corners and shadow.
Requires Pillow: pip install Pillow
"""
import sys
import os
from pathlib import Path

try:
    from PIL import Image, ImageDraw, ImageFilter
except ImportError:
    print("Error: Pillow is required. Install it with: pip install Pillow")
    sys.exit(1)

# Configuration
size = 1024
corner_radius = int(size * 0.22)  # ~22% corner radius (standard macOS)
shadow_offset = 10
shadow_blur = 30
shadow_opacity = 0.3

# Paths - resolve relative to project root
script_dir = Path(__file__).parent.resolve()
# From scripts/ -> desktop/ -> apps/ -> radium/ -> root/
project_root = script_dir / '../../../../'
input_path = project_root.resolve() / 'images/logo-apple.png'
output_path = script_dir.parent / 'app-icon.png'

print('Creating macOS-style icon...')
print(f'Input: {input_path}')
print(f'Output: {output_path}')

if not input_path.exists():
    print(f'Error: Input file not found at {input_path}')
    sys.exit(1)

try:
    # Load and resize source image
    img = Image.open(input_path)
    print(f'Source image: {img.size[0]}x{img.size[1]}')
    
    # Resize to target size while maintaining aspect ratio
    img = img.resize((size, size), Image.Resampling.LANCZOS)
    
    # Create shadow layer
    shadow_size = size + shadow_offset * 2
    shadow = Image.new('RGBA', (shadow_size, shadow_size), (0, 0, 0, 0))
    shadow_draw = ImageDraw.Draw(shadow)
    
    # Draw rounded rectangle for shadow
    shadow_draw.rounded_rectangle(
        [(shadow_offset, shadow_offset), 
         (size + shadow_offset, size + shadow_offset)],
        radius=corner_radius,
        fill=(0, 0, 0, int(255 * shadow_opacity))
    )
    
    # Apply blur to shadow
    shadow = shadow.filter(ImageFilter.GaussianBlur(radius=shadow_blur))
    
    # Create rounded corner mask
    mask = Image.new('L', (size, size), 0)
    mask_draw = ImageDraw.Draw(mask)
    mask_draw.rounded_rectangle(
        [(0, 0), (size, size)],
        radius=corner_radius,
        fill=255
    )
    
    # Apply mask to image
    img_masked = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    img_masked.paste(img, (0, 0))
    img_masked.putalpha(mask)
    
    # Composite shadow + masked image
    result = Image.new('RGBA', (shadow_size, shadow_size), (0, 0, 0, 0))
    result.paste(shadow, (0, 0), shadow)
    result.paste(img_masked, (shadow_offset, shadow_offset), img_masked)
    
    # Save result
    result.save(output_path, 'PNG', optimize=True)
    
    print(f'✅ macOS icon created successfully at {output_path}')
    print(f'   Size: {shadow_size}x{shadow_size}px')
    print(f'   Corner radius: {corner_radius}px')
    print(f'   Shadow: {shadow_offset}px offset, {shadow_blur}px blur')
    
except Exception as error:
    print(f'Error creating icon: {error}')
    import traceback
    traceback.print_exc()
    sys.exit(1)

